stackoversight/scraping/release_heap.py

```diff
@@ -0,0 +1,50 @@
+# Need a heap ofc
+import heapq
+# For release timers
+import threading
+# For the release queue
+from queue import Queue
+
+
+class ReleaseHeap(object):
+    """
+        elem_list must be a list of mutable tuples!
+        the left value needs to be comparable and incrementable/decrementable
+    """
+    def __init__(self, elem_list: list, time_sec: int):
+        # init the heap and queue and assign variables
+        heapq.heapify(elem_list)
+        self.heap = elem_list
+        self.time_sec = time_sec
+        self.release_queue = Queue()
+
+        # quick access to pair by it's value
+        self.hash_map = dict([(pair[1], pair) for pair in self.heap])
+
+    def __iter__(self):
+        return self.heap
+
+    def __next__(self):
+        return self.heap[0][1]
+
+    def capture(self):
+        # pop off the top of the heap and increment the first value of the pair
+        pair = heapq.heappop(self.heap)
+        pair[0] += 1
+        heapq.heappush(self.heap, pair)
+
+        # put the value into the queue
+        self.release_queue.put(pair[1])
+
+        # start the timer for release
+        threading.Timer(self.time_sec, self.release).start()
```

Comment:
How many times is this going to get called? Are we at risk of attempting to start thousands upon thousands of threads at once? Could we run out of memory and cause it to crash?